### ISSUE TITLE
Object properties change on editor after running game

### ISSUE THREAD

**Issue by carlos-montes-karasu** [godotengine/godot#90004]
### Tested versions

Reproducible in versions 4.2.1 Mono and 4.3.dev5 Mono (these are the only versions that I tried)

### System information

MacOS 14.4.1 - Godot 4.2.1 - Mobile

## Object Property changes in editor after game is running

I do not why, but when I play the game seeing the scene where the bug occur nothing weird happen, but If I play the game seeing another scene but that scene actually contains the scene that presents the bug, well, after closing the game, the bug scene with the animator, change the object properties, in this case the position

https://github.com/godotengine/godot/assets/79183176/a6ece758-ba62-41ac-b4f3-877dfedbfa6d

### Steps to reproduce

The way its happened to me is having two scenes one with an animator that modify control's nodes position, add that scene into another scene and run the having the second scene open in the editor

### Minimal reproduction project (MRP)

[issue_project.zip](https://github.com/godotengine/godot/files/14799547/issue_project.zip)

**Comment by carlos-montes-karasu** [comment]
I found something recently, by default the toggle "Reset on Save" is activated in the AnimationPlayer Node, I disabled the option and the bug did not happen again...

![image](https://github.com/godotengine/godot/assets/79183176/a0a52483-ebde-42a4-96ad-da2891c1d10c)

**Comment by AThousandShips** [comment]
This is indeed not a bug, it's by design and a feature that when saving the animation is reset

**Comment by carlos-montes-karasu** [comment]
I understand why that option is available, but when I run the game seeing another scene the editor makes change in the transform properties and breaks the UI layout, I think that is not intentional, right? or I am doing something wrong?

**Comment by AThousandShips** [comment]
That's true, my bad your description wasn't very clear, nor was it obvious from the video :) making it clear what you are dealing with is important

**Comment by carlos-montes-karasu** [comment]
Don't worry about it, I know that I not so good in english 😓 , sorry for the confuse
